--- a/src/modules/meson.build	2024-01-19 15:15:37.891619318 +0200
+++ b/src/modules/meson.build	2024-01-19 15:16:09.954484423 +0200
@@ -122,7 +122,7 @@
   'module-filter-chain/convolver.c'
 ]
 filter_chain_dependencies = [
-  mathlib, dl_lib, pipewire_dep, sndfile_dep, audioconvert_dep
+  mathlib, dl_lib, pipewire_dep, sndfile_dep
 ]
 
 pipewire_module_filter_chain = shared_library('pipewire-module-filter-chain',
@@ -175,7 +175,7 @@
   install : true,
   install_dir : modules_install_dir,
   install_rpath: modules_install_dir,
-  dependencies : [mathlib, dl_lib, pipewire_dep, audioconvert_dep],
+  dependencies : [mathlib, dl_lib, pipewire_dep],
 )
 
 build_module_jack_tunnel = jack_dep.found()
