diff --git a/configure b/configure
index 802d09e..b436d5e 100755
--- a/configure
+++ b/configure
@@ -26419,36 +26419,7 @@ if $PKG_CONFIG --uninstalled $ATK_PACKAGES; then
 else
 	gtk_save_LIBS="$LIBS"
         LIBS="$ATK_LIBS $LIBS"
-        cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
 
-/* Override any GCC internal prototype to avoid an error.
-   Use char because int might match the return type of a GCC
-   builtin and then its argument prototype would still apply.
-   The 'extern "C"' is for builds by C++ compilers;
-   although this is not generally supported in C code supporting it here
-   has little cost and some practical benefit (sr 110532).  */
-#ifdef __cplusplus
-extern "C"
-#endif
-char atk_object_get_type (void);
-int
-main (void)
-{
-return atk_object_get_type ();
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"
-then :
-  :
-else case e in #(
-  e) as_fn_error $? "
-                *** Cannot link to Accessibility Toolkit. Accessibility Toolkit is required
-                *** to build GTK+" "$LINENO" 5 ;;
-esac
-fi
 rm -f core conftest.err conftest.$ac_objext conftest.beam \
     conftest$ac_exeext conftest.$ac_ext
         LIBS="$gtk_save_LIBS"
diff --git a/configure.ac b/configure.ac
index c25528a..3fee115 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1338,17 +1338,6 @@ else
 ])
 fi
 
-if $PKG_CONFIG --uninstalled $ATK_PACKAGES; then
-        :
-else
-	gtk_save_LIBS="$LIBS"
-        LIBS="$ATK_LIBS $LIBS"
-        AC_TRY_LINK_FUNC(atk_object_get_type, : , AC_MSG_ERROR([
-                *** Cannot link to Accessibility Toolkit. Accessibility Toolkit is required
-                *** to build GTK+]))
-        LIBS="$gtk_save_LIBS"
-fi
-
 GTK_PACKAGES="atk cairo gdk-pixbuf-2.0 gio-2.0"
 if test "x$gdktarget" = "xx11"; then
   GTK_PACKAGES="$GTK_PACKAGES pangoft2"
diff --git a/configure~ b/configure~
index 23268e8..5adf7cd 100755
--- a/configure~
+++ b/configure~
@@ -24466,31 +24466,7 @@ if $PKG_CONFIG --uninstalled $ATK_PACKAGES; then
 else
 	gtk_save_LIBS="$LIBS"
         LIBS="$ATK_LIBS $LIBS"
-        cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
 
-/* Override any GCC internal prototype to avoid an error.
-   Use char because int might match the return type of a GCC
-   builtin and then its argument prototype would still apply.  */
-#ifdef __cplusplus
-extern "C"
-#endif
-char atk_object_get_type ();
-int
-main ()
-{
-return atk_object_get_type ();
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  :
-else
-  as_fn_error $? "
-                *** Cannot link to Accessibility Toolkit. Accessibility Toolkit is required
-                *** to build GTK+" "$LINENO" 5
-fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
         LIBS="$gtk_save_LIBS"
