diff --git a/src/XFileImage.cpp b/src/XFileImage.cpp
index 01a045c..515f34a 100644
--- a/src/XFileImage.cpp
+++ b/src/XFileImage.cpp
@@ -2873,6 +2872,0 @@ int main(int argc, char* argv[])
-    // Icon path not found
-    if (!iconpathfound)
-    {
-        MessageBox::error(application, BOX_OK, _("Error loading icons"), _("Icon path doesn't exist, icon theme was set back to default. Please check your icon path!") );
-    }
-
diff --git a/src/XFilePackage.cpp b/src/XFilePackage.cpp
index 28cd76e..c137c85 100644
--- a/src/XFilePackage.cpp
+++ b/src/XFilePackage.cpp
@@ -885,6 +884,0 @@ int main(int argc, char* argv[])
-    // Icon path not found
-    if (!iconpathfound)
-    {
-        MessageBox::error(application, BOX_OK, _("Error loading icons"), _("Icon path doesn't exist, icon theme was set back to default. Please check your icon path!") );
-    }
-
diff --git a/src/XFileWrite.cpp b/src/XFileWrite.cpp
index 870eeeb..2590974 100644
--- a/src/XFileWrite.cpp
+++ b/src/XFileWrite.cpp
@@ -351,6 +350,0 @@ int main(int argc, char* argv[])
-    // Icon path not found
-    if (!iconpathfound)
-    {
-        MessageBox::error(application, BOX_OK, _("Error loading icons"), _("Icon path doesn't exist, icon theme was set back to default. Please check your icon path!") );
-    }
-
diff --git a/src/main.cpp b/src/main.cpp
index e8e58ea..9c0a4ab 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -563,6 +562,0 @@ int main(int argc, char* argv[])
-    // Icon path not found
-    if (!iconpathfound)
-    {
-        MessageBox::error(application, BOX_OK, _("Error loading icons"), _("Icon path doesn't exist, icon theme was set back to default. Please check your icon path!") );
-    }
-
