diff --git a/dom/base/nsGlobalWindow.cpp b/dom/base/nsGlobalWindow.cpp
index 7828f5bd52..e8c9522f11 100644
--- a/platform/dom/base/nsGlobalWindow.cpp
+++ b/platform/dom/base/nsGlobalWindow.cpp
@@ -3543,7 +3543,7 @@ nsGlobalWindow::GetEventTargetParent(EventChainPreVisitor& aVisitor)
 
   aVisitor.mCanHandle = true;
   // Middle/right click shouldn't dispatch click event, use auxclick to instead.
-  if (mDoc->IsXULDocument()) {
+  if (mDoc != NULL && mDoc->IsXULDocument()) {
     aVisitor.mForceContentDispatch = true; //FIXME! Bug 329119
   }
   if (msg == eResize && aVisitor.mEvent->IsTrusted()) {
